torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1306,6 +1306,30 @@ at::Tensor XLANativeFunctions::expand(const at::Tensor& self,
       bridge::GetXlaTensor(self), torch::lazy::ToVector<int64_t>(size)));
 }
 
+at::Tensor XLANativeFunctions::expand_symint(const at::Tensor& self,
+                                             c10::SymIntArrayRef size,
+                                             bool implicit) {
+  XLA_FN_COUNTER("xla::");
+  SymIntElements size_elements = SymIntElements(size);
+  // Replace -1 concrete int dim with the true shape value
+  std::vector<c10::SymInt> _sizes = torch::lazy::ToVector<c10::SymInt>(size);
+  int64_t num_new_dimensions = _sizes.size() - self.dim();
+  std::vector<int64_t> padded_self(num_new_dimensions, 0);
+  padded_self.insert(padded_self.end(), self.sizes().begin(),
+                     self.sizes().end());
+  for (const auto idx : c10::irange(_sizes.size())) {
+    if (!_sizes[idx].is_symbolic() && _sizes[idx].expect_int() == -1) {
+      size_elements.SetUpperBound(idx, padded_self[idx]);
+    }
+  }
+  std::vector<torch::lazy::Shape> dynamic_shapes_ =
+      torch::lazy::compute_shape_expand(self, size, implicit);
+  return bridge::AtenFromXlaTensor(XLATensor::expand_symint(
+      bridge::GetXlaTensor(self), size_elements.GetNodes(),
+      size_elements.GetUpperBounds(), size_elements.GetDynamicDims(),
```

Comment:
question: any reason not to pass `size_elements` directly?

Reply:
I can as discussed below.